core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -3394,9 +3397,12 @@ public void cleanup() {
                         dirty = true;
                         log.info("Removed transaction {} from pending pool during cleanup.", tx.getTxId());
                     } else {
+                        String reason = String.format("%s%s%s", isAnyOutputSpent ? "spent partially" :"",
+                                isAnyOutputSpent && isISLocked ? " and " : "",
+                                isISLocked ? "instant send locked" : "");
                         log.info(
-                                "Cannot remove transaction {} from pending pool during cleanup, as it's already spent partially.",
-                                tx.getTxId());
+                                "Cannot remove transaction {} from pending pool during cleanup, as it's already {}.",
+                                tx.getTxId(), reason);
```

Comment:
This needs to be removed from this PR.

Reply:
Is this still relevant?

Reply:
Unfortunately, I forgot why I put this comment.  It is OK to leave it in since it only changes how some information is logged.

Reply:
Actually, I put this first comment because it hasn't been tested and there are no unit tests for it.  We should remove this until a another time when it can be tested.

The idea was to prevent v2 transactions from being removed from the wallet.  In Dash Wallet, another solution was used.